crates/apollo-router/src/configuration/otlp/mod.rs

```diff
@@ -80,6 +80,37 @@ impl ExportConfig {
     }
 }
 
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(transparent)]
+pub struct EndpointUrl {
+    #[serde(deserialize_with = "endpoint_url")]
+    endpoint: Url,
+}
+
+fn endpoint_url<'de, D>(deserializer: D) -> Result<Url, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let mut buf = String::deserialize(deserializer)?;
+
+    // support the case of a IP:port endpoint
+    if buf.parse::<std::net::SocketAddr>().is_ok() {
+        buf = format!("https://{}", buf);
```

Comment:
should we document somewhere that we're falling back to https ?

Reply:
in https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/otlpexporter/README.md#getting-started "By default, TLS is enabled" so it's expected. But I agree that it should be in the documentation.
I'm planning to rewrite the part of the documentation about telemetry once this week's pull requests have settled a bit